File: examples/docs-code-snippets/src/main/scala/examples/scala/application/FireEvent.scala
package examples.scala.application

import com.ing.baker.runtime.scaladsl.{Baker, EventInstance}
import examples.scala.events.OrderPlaced

class FireEvent(val baker: Baker) {

  def example(recipeInstanceId: String, orderPlaced: OrderPlaced): Unit = {
    val eventInstance = EventInstance.unsafeFrom(orderPlaced)

    val eventResolutions = baker.fireEvent(recipeInstanceId, eventInstance)
    val sensoryEventStatus = eventResolutions.resolveWhenReceived
    val sensoryEventResult = eventResolutions.resolveWhenCompleted
  }
}

File: examples/docs-code-snippets/src/main/kotlin/examples/kotlin/application/FireEvent.kt
package examples.kotlin.application

import com.ing.baker.runtime.javadsl.EventInstance
import com.ing.baker.runtime.kotlindsl.Baker
import examples.kotlin.events.OrderPlaced

class FireEvent(private val baker: Baker) {

    suspend fun example(recipeInstanceId: String, orderPlaced: OrderPlaced) {
        val orderPlacedEvent = EventInstance.from(orderPlaced)

        val eventResolutions = baker.fireEvent(recipeInstanceId, orderPlacedEvent)
        val sensoryEventStatus = eventResolutions.resolveWhenReceived.await()
        val sensoryEventResult = eventResolutions.resolveWhenCompleted.await()
    }
}